### PR TITLE
GP2-575 additional news page functionality

### PR DESCRIPTION
**Pull request by AkosuaA**

### CODE REVIEW

File: apps/gp2-frontend/src/news/NewsList.tsx [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
-          currentPageIndex={currentPage}
-          renderPageHref={renderPageHref}
-        >
-          {items.map((news) => (
-            <NewsItem key={news.id} {...news} />
-          ))}
-        </ResultList>
-      </Frame>
-    </NewsListPage>
+  const result = useNews({
```
Comment (markcsims) [inline apps/gp2-frontend/src/news/NewsList.tsx:13]:
```suggestion
  const { items, total } = useNews({
```

File: apps/gp2-frontend/src/news/__tests__/state.test.tsx
Comment (markcsims) [inline apps/gp2-frontend/src/news/__tests__/state.test.tsx:1]: Do we need to test "state.tsx" - I thought the tests around the News Component should cover this?

File: apps/gp2-server/test/data-providers/news.data-provider.test.ts [hunk 2]
```diff
@@ -154,6 +154,71 @@ describe('News data provider', () => {
...
+          getContentfulNewsGraphqlResponse(),
+        );
+
+        const result = await newsDataProvider.fetch({
+          filter: { type: ['news'] },
+        });
+
+        expect(result).toEqual(getListNewsDataObject());
+        expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+          expect.anything(),
```
Comment (markcsims) [inline apps/gp2-server/test/data-providers/news.data-provider.test.ts:170]: You could possibly replace expect.anything() in this spec with gp2Contentful.FETCH_NEWS -

**Review by markcsims — CHANGES_REQUESTED** [review]

File: apps/gp2-frontend/src/news/__tests__/state.test.tsx
Comment (AkosuaA) [reply apps/gp2-frontend/src/news/__tests__/state.test.tsx:1]: There were certain portions that weren't covered. I wasn't sure how to mimic those in the News component tests so I tested them directly

File: apps/gp2-frontend/src/news/__tests__/state.test.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      }),
+      {
+        wrapper: RecoilRoot,
+      },
+    );
+
+    const setNewsList = result.current.newsListState[1];
+    const setItemState = result.current.newsItemState;
+
+    await act(() => mockGetNews(options, '').then(setNewsList));
```
Comment (markcsims) [inline apps/gp2-frontend/src/news/__tests__/state.test.tsx:70]: We are executing a mock

Also mixing Promise chains with async...await

**Review by markcsims — CHANGES_REQUESTED** [review]

**Review by markcsims — APPROVED** [review]

**Pull request merged**